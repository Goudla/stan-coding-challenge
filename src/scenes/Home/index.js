import React, { Component } from 'react';
import Helmet from 'react-helmet';
import { Link } from "react-router-dom";

// Components
import Col from '../../components/Col';
import Container from '../../components/Container';
import FlexGrid from '../../components/FlexGrid';
import PageContainer from '../../components/PageContainer';
import SubNavbar from '../../components/SubNavbar';
import SubNavbarHeading from '../../components/SubNavbarHeading';
import Title from '../../components/Title';

/**
 *  The home scene of the application
 * @extends Component
 */
class Home extends Component<*> {
  render() {
    return (
      <div>
        <Helmet>
          <title>Home</title>
        </Helmet>
        <SubNavbar>
          <Container>
            <SubNavbarHeading>Popular Titles</SubNavbarHeading>
          </Container>
        </SubNavbar>
        <PageContainer>
          <FlexGrid>
            <Col>
              <Link
                to="/series">
                <Title>Popular Series</Title>
              </Link>
            </Col>
            <Col>
              <Link to="/movies">
                <Title>Popular Movies</Title>
              </Link>
            </Col>
          </FlexGrid>
        </PageContainer>
      </div>
    );
  }
}

export default Home;
