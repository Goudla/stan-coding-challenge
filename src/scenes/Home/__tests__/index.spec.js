import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

// Scenes
import Home from '../';

describe('<Home />', () => {
  it('Should render', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Home />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
