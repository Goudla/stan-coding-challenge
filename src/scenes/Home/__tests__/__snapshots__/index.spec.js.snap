// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Should render 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Home
    </title>
  </HelmetWrapper>
  <SubNavbar>
    <Container>
      <SubNavbarHeading>
        Popular Titles
      </SubNavbarHeading>
    </Container>
  </SubNavbar>
  <PageContainer>
    <FlexGrid>
      <Col>
        <Link
          to="/series"
        >
          <Title>
            Popular Series
          </Title>
        </Link>
      </Col>
      <Col>
        <Link
          to="/movies"
        >
          <Title>
            Popular Movies
          </Title>
        </Link>
      </Col>
    </FlexGrid>
  </PageContainer>
</div>
`;
